src/main/java/org/betonquest/betonquest/objectives/EnchantObjective.java

```diff
@@ -17,22 +16,26 @@
 
 import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 
 /**
  * Requires the player to enchant an item.
  */
 @SuppressWarnings("PMD.CommentRequired")
-public class EnchantObjective extends Objective implements Listener {
+public class EnchantObjective extends CountingObjective implements Listener {
 
     private final QuestItem item;
-    private final List<EnchantmentData> enchantments;
+    private final List<EnchantmentData> desiredEnchantments;
 
     public EnchantObjective(final Instruction instruction) throws InstructionParseException {
         super(instruction);
-        template = ObjectiveData.class;
+        targetAmount = instruction.getInt(instruction.getOptional("amount"), 1);
+        if (targetAmount <= 0) {
+            throw new InstructionParseException("Amount cannot be less than 1.");
+        }
         item = instruction.getQuestItem();
-        enchantments = instruction.getList(EnchantmentData::convert);
-        if (enchantments.isEmpty()) {
+        desiredEnchantments = instruction.getList(EnchantmentData::convert);
+        if (desiredEnchantments.isEmpty()) {
             throw new InstructionParseException("Not enough arguments");
```

Comment:
Can we also be a bit more explicit here now, that there are no enchantments

Reply:
Good catch. I think I will use the opportunity to allow "nothing" as "any enchantment".

Reply:
How would you even give an empty list, this would then consume the next (labeled) argument if I remember correctly: `enchant item events:event_a,event_b` would use `events:event_a` as the first enchantment and `event_b` as the second.

Reply:
True, this is sadly not possible.